File: code_snippets/ts/src/develop/durable_timers.ts
import * as restate from "@restatedev/restate-sdk";

const service = restate.service({
  name: "DurableTimers",
  handlers: {
    greet: async (ctx: restate.Context, name: string) => {
      // <start_here>
      await ctx.sleep({ seconds: 10 });
      // <end_here>
    },
  },
});

File: code_snippets/go/develop/durabletimers.go
package develop

import (
	"time"

	restate "github.com/restatedev/sdk-go"
)

type DurableTimers struct{}

func (DurableTimers) Greet(ctx restate.Context, name string) error {
	// <start_here>
	if err := restate.Sleep(ctx, 10*time.Second); err != nil {
		return err
	}
	// <end_here>
	return nil
}
